src/Agent.Worker/Build/GitSourceProvider.cs

```diff
@@ -25,18 +25,26 @@ public class GitSourceProvider : SourceProvider, ISourceProvider
 
         public override string RepositoryType => WellKnownRepositoryTypes.Git;
 
-        public virtual async Task GetSourceAsync(IExecutionContext executionContext, ServiceEndpoint endpoint, CancellationToken cancellationToken)
+        public virtual async Task GetSourceAsync(
+            IExecutionContext executionContext,
+            ServiceEndpoint endpoint,
+            CancellationToken cancellationToken)
         {
             Trace.Entering();
+            // Validate args.
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
             ArgUtil.NotNull(endpoint, nameof(endpoint));
 
             executionContext.Output($"Syncing repository: {endpoint.Name} (Git)");
             _gitCommandManager = HostContext.GetService<IGitCommandManager>();
             await _gitCommandManager.LoadGitExecutionInfo(executionContext);
 
-            string targetPath = executionContext.Variables.Get(Constants.Variables.Build.SourcesDirectory);
-            string sourceBranch = executionContext.Variables.Get(Constants.Variables.Build.SourceBranch);
-            string sourceVersion = executionContext.Variables.Get(Constants.Variables.Build.SourceVersion);
+            string targetPath;
+            string sourceBranch;
+            string sourceVersion;
+            endpoint.Data.TryGetValue(Constants.Variables.Build.SourcesDirectory, out targetPath);
```

Comment:
I thought about it. But then we would be duplicating strings when infact we dont have to. Then i thought about renaming the Variables class to something generic so that the strings could be used at multiple locations. But that would be a lot of unnecessary work. So i was stuck with it.
My only concern is to avoid DRY violation wherever we can. But if you stongly feel that we should take this comment then let me know.